### DISCUSSION TITLE
Make it possible to speed up loading time by using uv

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [pyscript/pyscript#2122] [Proposals]
I can see that Maarten A. Breddals speeds up loading of Pyodide based data apps at py.cafe by using `uv`. Please enable this possibility in Pyscript too.

See https://www.linkedin.com/posts/maartenbreddels_dash-python-pyodide-activity-7206296573264195584-dgAO?utm_source=share&utm_medium=member_desktop

![image](https://github.com/user-attachments/assets/2d061246-670e-4b52-a0f3-ffdda2ac842f)

https://github.com/user-attachments/assets/ec3f1671-0cee-4a88-bb27-c20d7ff8c3db

**Comment by WebReflection** [comment]
what is *uv* ? There is a long standing issue we need to tackle first to allow caching of the env to bootstrap future reload of the same env (read: same packages required sent as already resolved links to download) + this might work for PyScirpt .com but not really as well (on first run) for examples that are outside PyScript .com space (because it can pre-parse the config and create once that already resolved dependency).

That being said, I see a lot of time waiting for PyCafe to just show something so I think the fast feeling once that is done is hidden behind the initial loading screen that doesn't look super fast to me.

Curious to learn more around *uv* though, thanks.

**Reply by MarcSkovMadsen** [reply]
Uv is a pip alternative. See https://astral.sh/blog/uv

**Reply by WebReflection** [reply]
then I am afraid I am not sure how/if Pyodide supports that ... I will investigate though ... we just use Pyodide as it is and we trust it offers its best by default.
